jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -260,6 +263,31 @@ object XGBoost extends Serializable {
     }
   }
 
+  private def repartitionForTrainingGroup(
+      trainingData: RDD[XGBLabeledPoint], nWorkers: Int): RDD[Array[XGBLabeledPoint]] = {
+    val normalGroups = trainingData.mapPartitions(
+      new LabeledPointGroupIterator(_)).filter(!_._1).map(_._2)
+    val edgeGroups = trainingData.mapPartitions(
```

Comment:
changed the repartitionForTrainingGroup(..) to stitch the chunks of a group from one or more partitions.